### PR TITLE
Also run consul servers on oc*

### PR DESCRIPTION
**Pull request by sorenh**

### CODE REVIEW

**Comment by sorenh** [comment]
please test this

**Comment by sorenh** [comment]
On a scale from 0 to wow-that-was-easy, this rates a solid "doh". The leader election didn't work out well at all.

**Comment by sorenh** [comment]
please test this

**Comment by sorenh** [comment]
please test this

**Comment by sorenh** [comment]
please test this

**Comment by sorenh** [comment]
please test this

**Comment by bodepd** [comment]
is it that easy :)

**Comment by sorenh** [comment]
please test this

**Comment by sorenh** [comment]
What you can't see here is that I'm building a new consul package out of 0.5.0rc1 and testing it. To avoid blocking AT in case this new version doesn't work as expected, I'm having to name it differently and it's taking a few attempts to get all that right.

**Comment by sorenh** [comment]
please test this

File: Puppetfile [hunk 1]
```diff
@@ -157,7 +157,7 @@ mod 'puppet-modules/common',
   :ref => 'origin/master'
 
 mod 'solarkennedy/consul',
-  :git => "#{base_url}/solarkennedy/puppet-consul",
+  :git => "#{base_url}/JioCloud/puppet-consul",
```
Comment (bodepd) [inline Puppetfile]: any reason this did not get submitted upstream?

File: hiera/data/common.yaml [hunk 2]
```diff
@@ -788,3 +788,5 @@ tempest::provision::networkname: "%{hiera('tempest::fixed_network_name')}"
 
 apache::trace_enable: 'Off'
 apache::mod::ssl::ssl_cipher: "HIGH:MEDIUM:!aNULL:!MD5:!RC4"
+
+consul::package_name: consul0.5.0
```
Comment (bodepd) [inline hiera/data/common.yaml]: do we need to move to 0.5.0 as a part of this effort? Is there an exact feature that we need?

**Comment by bodepd** [comment]
please test this

**Comment by bodepd** [comment]
please test this

**Comment by bodepd** [comment]
the cluster forms correctly:

````
Node                                           Address         Status  Type    Build     Protocol
gcp1-testjenkins-puppet-rjil-gate-792          10.0.0.24:8301  alive   client  0.5.0rc1  2
ct1-testjenkins-puppet-rjil-gate-792           10.0.0.22:8301  alive   client  0.5.0rc1  2
haproxy1-testjenkins-puppet-rjil-gate-792      10.0.0.18:8301  alive   client  0.5.0rc1  2
bootstrap1-testjenkins-puppet-rjil-gate-792    10.0.0.28:8301  alive   server  0.5.0rc1  2
stmonleader1-testjenkins-puppet-rjil-gate-792  10.0.0.20:8301  alive   client  0.5.0rc1  2
stmon2-testjenkins-puppet-rjil-gate-792        10.0.0.21:8301  alive   client  0.5.0rc1  2
stmon1-testjenkins-puppet-rjil-gate-792        10.0.0.27:8301  alive   client  0.5.0rc1  2
oc1-testjenkins-puppet-rjil-gate-792           10.0.0.23:8301  alive   server  0.5.0rc1  2
cp2-testjenkins-puppet-rjil-gate-792           10.0.0.26:8301  alive   client  0.5.0rc1  2
ocdb1-testjenkins-puppet-rjil-gate-792         10.0.0.19:8301  alive   server  0.5.0rc1  2
cp1-testjenkins-puppet-rjil-gate-792           10.0.0.25:8301  alive   client  0.5.0rc1  2
````

but no services are registering after the first run. I can see that this is b/c validation and puppet are in a critical state. I think this is a difference between consul 4 and 5. The fix for this is the same fix we need to do anyways, bind these checks to services. I will update this patch to include code that does that...

**Comment by bodepd** [comment]
replaced with #457

**Pull request closed without merging**